src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -130,6 +130,22 @@ public Page<AdvertisementTitleDto> findAdvertisementsHavingCategory(
         return advertisementService.findByCategoryId(categoryId, PageRequest.of(page, size));
     }
 
+    @GetMapping(ApiKey.ADV_SUBCATEGORY_RANDOM)
+    @Operation(summary = "Find 4 random advertisement with same subcategory without request advertisement and return them as a result")
+    @ApiResponse(responseCode = "200", description = "OK")
+    public List<AdvertisementTitleDto> findRandom4AdvertisementWithSameSubcategory(
```

Comment:
It would be better to create 2 separate endpoints: 1 - random, and 2 - pageable.
Her you get **page** and **size** values but inside the code generate the random page instead and that isn't consistent.